xray/commands/audit/javautils.go

```diff
@@ -84,22 +85,28 @@ func runScanGraph(modulesDependencyTrees []*services.GraphNode, serverDetails *c
 		if err != nil {
 			return err
 		}
-
-		if err = xrutils.WriteJsonResults(scanResults, tempDirPath); err != nil {
-			return err
+		if printFormat == Table {
+			if err = xrutils.WriteJsonResults(scanResults, tempDirPath); err != nil {
+				return err
+			}
 		}
 		violations = append(violations, scanResults.Violations...)
 		vulnerabilities = append(vulnerabilities, scanResults.Vulnerabilities...)
 		licenses = append(licenses, scanResults.Licenses...)
-	}
-	fmt.Println("The full scan results are available here: " + tempDirPath)
-	if includeVulnerabilities {
-		xrutils.PrintVulnerabilitiesTable(vulnerabilities, false)
-	} else {
-		err = xrutils.PrintViolationsTable(violations, false)
-	}
-	if includeLicenses {
-		xrutils.PrintLicensesTable(licenses, false)
+		results = append(results, *scanResults)
+	}
+	if printFormat == Table {
+		fmt.Println("The full scan results are available here: " + tempDirPath)
+		if includeVulnerabilities {
+			xrutils.PrintVulnerabilitiesTable(vulnerabilities, false)
+		} else {
+			err = xrutils.PrintViolationsTable(violations, false)
+		}
+		if includeLicenses {
+			xrutils.PrintLicensesTable(licenses, false)
+		}
+	} else if printFormat == Json {
```

Comment:
There are only 2 options, I think we should block illegal inputs from the CLI code
```suggestion
	} else {
```